lib/ecto/query/builder.ex

```diff
@@ -584,10 +584,15 @@ defmodule Ecto.Query.Builder do
     {:{}, [], [dot, [], []]}
   end
 
-  defp escape_field!({kind, _, [atom]}, field, _vars)
+  defp escape_field!({kind, _, [value]}, field, _vars)
        when kind in [:as, :parent_as] do
-    atom  = quoted_atom!(atom, "#{kind}/1")
-    as    = {:{}, [], [kind, [], [atom]]}
+    value =
```

Comment:
This is just a temporary measure, unsure how it ought to be handled.

I can see why we might want to keep people "on the rails", e.g the only way that you can add a binding that isn't an atom would be if you used `^value` or something like that. Makes no difference to me. We could also just remove the necessity to pin anything in `as/1` and `parent_as/1`, but that would break things on upgrade.